### PR TITLE
Page routing

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>
Create tsx files for necessary pages in app
Initial boiler plate code
Implemented routing using <Route /> tags and react router for navigating through the website using the url
Implemented NavLink to navigate across <Route /> pages

<what will be done in later PRs and not included here>
Implementing the "on-click" functionality for the navigation bar

## Milestones
<the milestones or stories/features that this works towards>
Navigation through different pages on the website

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
React Router Documentation: https://reactrouter.com/start/declarative/routing
Webpage on nested routes (inspiration for page not found error routing): https://www.robinwieruch.de/react-router-nested-routes/
React router navlink documentation: https://reactrouter.com/start/declarative/navigating
Styling navlink: https://medium.com/@alexanie_/navlink-component-in-react-router-b83f4a11794f

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
URL Routes: https://www.loom.com/share/4884b67b919b4266a6838e2e372ccde1
Navigation Bar: https://www.loom.com/share/d663b2b91cad469aa4b070c07aed25d8

### CODE REVIEW

File: frontend/src/pages/RecipesPage.tsx [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
-            <Outlet />
-        </div>
-    )
-}
+  const [navOpen, setNavOpen] = useState<boolean>(false);
+
+  return (
+    <div>
+      <button onClick={() => setNavOpen((prev) => !prev)}>
+        <FontAwesomeIcon
```
Comment (radhikadhomse) [inline frontend/src/pages/RecipesPage.tsx:13]: you could probably remove the onClick from FontAwesomeIcon to avoid double toggling the onclick

File: frontend/src/pages/RecipesPage.tsx [hunk 2]
```diff
@@ -1,16 +1,27 @@
-import '../styles/RecipesPage.css'
-import { Outlet } from 'react-router';
-import NavBar from '../components/NavBar';
-
+import "../styles/RecipesPage.css";
+import { Outlet } from "react-router";
+import NavBar from "../components/NavBar";
+import { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBars, faTicket } from "@fortawesome/free-solid-svg-icons";
```
Comment (radhikadhomse) [inline frontend/src/pages/RecipesPage.tsx:6]: is faTicket used?

File: frontend/src/pages/RecipesPage.tsx (hunk 1 above)
Comment (Brianna-Flo) [reply frontend/src/pages/RecipesPage.tsx:12]: Removed!

**Review by paulshaoyuqiao — APPROVED** [review]
looks good to me!
